### PR TITLE
Feature/282 readme update

### LINKED ISSUE #282: Improve documentation
Improve documentation: 
- How it works
- How ui looks like

### PR DESCRIPTION
**Pull request by jozefbakus**
Impl: #282

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,8 +1,168 @@
...
+
+# What is Hyperdrive-trigger?
+**Hyperdrive-trigger** is a **Event based workflow manager and scheduler**.
+
+Workflow is defined via graphical interface and consists of three parts: **details**, **sensor** and **jobs**:
+- **details** - General workflow's information (workflow name, project name and whether is workflow active)
+- **sensor** - Definition of workflow trigger, describes when workflow will be executed. Sensors that could be used:
+  - *Kafka* - waits for kafka message with specific message 
+  - *Time* - time based trigger, cron-quartz expression or user friendly time recurring definition can be used
+  - *Recurring* - worklfow is triggered always when previous execution is finished
```
Comment (kevinwallimann) [inline README.md]: Typo worklfow

File: README.md [hunk 2]
```diff
@@ -1,8 +1,168 @@
...
+    - [Docker image](#docker-image)
+    - [Web Application Archive](#web-application-archive)
+- [User Interface](#user-interface)
+- [How to contribute](#how-to-contribute)
+<!-- tocstop -->
+
+# What is Hyperdrive-trigger?
+**Hyperdrive-trigger** is a **Event based workflow manager and scheduler**.
+
+Workflow is defined via graphical interface and consists of three parts: **details**, **sensor** and **jobs**:
```
Comment (kevinwallimann) [inline README.md]: Missing articles: A workflow is defined via the graphical interface

File: README.md [hunk 3]
```diff
@@ -1,8 +1,168 @@
...
+- **details** - General workflow's information (workflow name, project name and whether is workflow active)
```
Comment (kevinwallimann) [inline README.md]: I would use uppercase for details, sensor and jobs in the enumeration

File: README.md [hunk 4]
```diff
@@ -1,8 +1,168 @@
...
+- **sensor** - Definition of workflow trigger, describes when workflow will be executed. Sensors that could be used:
+  - *Kafka* - waits for kafka message with specific message 
+  - *Time* - time based trigger, cron-quartz expression or user friendly time recurring definition can be used
+  - *Recurring* - worklfow is triggered always when previous execution is finished 
+- **jobs** - list of jobs, defined as a linear directed acyclic graph. Supported jobs types:
```
Comment (kevinwallimann) [inline README.md]: linear directed acylic graph. This is quite misleading :) It's just a list of jobs. I would remove `defined as a linear directed acyclic graph`

Comment (kevinwallimann) [inline README.md]: `jobs types` -> `job types`

File: README.md [hunk 5]
```diff
@@ -1,8 +1,168 @@
...
+
+Workflow is defined via graphical interface and consists of three parts: **details**, **sensor** and **jobs**:
+- **details** - General workflow's information (workflow name, project name and whether is workflow active)
+- **sensor** - Definition of workflow trigger, describes when workflow will be executed. Sensors that could be used:
+  - *Kafka* - waits for kafka message with specific message 
+  - *Time* - time based trigger, cron-quartz expression or user friendly time recurring definition can be used
+  - *Recurring* - worklfow is triggered always when previous execution is finished 
+- **jobs** - list of jobs, defined as a linear directed acyclic graph. Supported jobs types: 
+  - *Spark* - spark job deployed to Apache Hadoop YARN
+  - *Shell* - shell job
```
Comment (kevinwallimann) [inline README.md]: Now it's `Generic Spark Job` and `Generic Shell Job`

Maybe some description for the shell-job, e.g. Job executing a user-defined shell script.

File: README.md [hunk 6]
```diff
@@ -1,8 +1,168 @@
...
+
+User interface also provide nice visualization of running workflow with ability to monitor and troubleshoot execution.
```
Comment (kevinwallimann) [inline README.md]: Articles: _The_ user interface provides _a_ nice visualization of running workflow_s_ with _the_ ability to monitor and troubleshoot execution_s_.

Omit _nice_. It's not a descriptive, but evaluative word.

You could add a link to the screenshots

File: README.md [hunk 7]
```diff
@@ -1,8 +1,168 @@
...
+
+# Requirements
+Tested with:
+|              |                        |
+| ------------ | ---------------------- | 
+| OpenJDK      | 1.8.0_25               |
+| Scala        | 2.11                   |
+| Maven        | 3.5.4+                 |
```
Comment (kevinwallimann) [inline README.md]: Why the plus? You can't really say you have tested with 3.5.4+ (unless you immediately test as soon as a new version is released)

File: README.md [hunk 8]
```diff
@@ -1,8 +1,168 @@
...
+sparkYarnSink.additionalConfs.spark.executor.instances=2
+sparkYarnSink.additionalConfs.spark.executor.cores=2
+sparkYarnSink.additionalConfs.spark.executor.memory=2g
+sparkYarnSink.additionalConfs.spark.yarn.keytab=
+sparkYarnSink.additionalConfs.spark.yarn.principal=
+sparkYarnSink.additionalConfs.spark.shuffle.service.enabled=true
+sparkYarnSink.additionalConfs.spark.dynamicAllocation.enabled=true
+```
+```
+#Postgresql properties for connection to trigger metastore
```
Comment (kevinwallimann) [inline README.md]: I think `database` is more appropriate than `metastore`.

File: README.md [hunk 9]
```diff
@@ -1,8 +1,168 @@
...
+| Angular CLI  | 9.0.3                  | 
+
+> Note: Docker is not required.
+
+# How to build and run
+## Application properties
+Before building application, update of application properties is required. Application properties templeate file could be find at `src/main/resources/application.properties`. Application properties that could be adjusted: 
+
+```
+# Version of application.
```
Comment (kevinwallimann) [inline README.md:58]: Do we really need to print the whole application.properties here?

File: README.md [hunk 10]
```diff
@@ -1,8 +1,168 @@
...
+| Tomcat       | 9.0.24+                | 
+| Docker       | 2.3.0.5                | 
+| NPM          | 6.14.4                 | 
+| Angular CLI  | 9.0.3                  | 
+
+> Note: Docker is not required.
+
+# How to build and run
+## Application properties
+Before building application, update of application properties is required. Application properties templeate file could be find at `src/main/resources/application.properties`. Application properties that could be adjusted:
```
Comment (kevinwallimann) [inline README.md]: You don't have to adjust the application properties file before building it. You can also start the app with an external properties file or update it directly in the jar file.

Typos: `templeate`

File: README.md [hunk 11]
```diff
@@ -1,8 +1,168 @@
...
+db.url=jdbc:postgresql://
+db.user=
+db.password=
+db.keepAliveConnection=true
+db.connectionPool=HikariCP
+db.numThreads=4
+```
+
+## Embedded Tomcat
+#### Tested with :
```
Comment (kevinwallimann) [inline README.md]: Do we need to list these dependencies here again? What's the difference to the requirements listed above

File: README.md [hunk 12]
```diff
@@ -1,8 +1,168 @@
...
+ - **Maven 3.5.4+**
+ - **NPM 6.14.4**
+ - **Angular CLI 9.0.3**
+ - **OpenJDK 1.8.0_25**
+ 
+For development purposes, hyperdrive-trigger can be executed as a application with embedded tomcat. Embedded tomcat version can be found in **feature/embedded-tomcat-2** branch with latest changes from develop branch.
```
Comment (kevinwallimann) [inline README.md]: Articles: a _n_ application with _an_ embedded tomcat.

Maybe rephrase the second sentence
Please check out branch **feature/embedded-tomcat-2** to use it.

File: README.md [hunk 13]
```diff
@@ -1,8 +1,168 @@
...
+
+To build executable jar and execute it use following commands:
```
Comment (kevinwallimann) [inline README.md]: Articles: 
- To build _an_ executable
- use _the_ following
Comma after it: execute it, use

File: README.md [hunk 14]
```diff
@@ -1,8 +1,168 @@
...
+- Package jar: `mvn clean package` or without tests `mvn clean package -DskipTests`
+- Execute it: `java -jar ./target/hyperdrive-trigger-<VERSION>.jar`
+
+Access the application at 
+```
+http://localhost:7123/#
+```
+
+For fast, local and iterative front end development you can run ui separatly using following commands:
```
Comment (kevinwallimann) [inline README.md]: I would rephrase:
For fast, local and iterative front end development, the UI can be run using a live development server with the following commands.

Do we really need to write `fast, local and iterative` Sounds like we're very proud of that :)

File: README.md [hunk 15]
```diff
@@ -1,8 +1,168 @@
...
+- Package jar: `mvn clean package` or without tests `mvn clean package -DskipTests`
+- Execute it: `java -jar ./target/hyperdrive-trigger-<VERSION>.jar`
+
+Access the application at 
+```
+http://localhost:7123/#
+```
+
+For fast, local and iterative front end development you can run ui separatly using following commands: 
+- Install required packages: `cd ui & npm install`
```
Comment (kevinwallimann) [inline README.md]: There should be two ampersands. Single ampersand runs commands in parallel

File: README.md [hunk 16]
```diff
@@ -1,8 +1,168 @@
...
+- Strat front end application: `cd ui & ng serve` or `cd ui & npm start`
```
Comment (kevinwallimann) [inline README.md]: Typo `Strat`

File: README.md [hunk 17]
```diff
@@ -39,5 +199,35 @@ docker volume prune
...
 This removes stopped containers and volumes that are not referenced by any containers.
 
-## Testdata
-`mvn test -Ptestdata` creates testdata in `localhost:5432/hyperdriver`, assuming the DB-user `hyperdriver` exists.
+## Web Application Archive
+#### Tested with :
+ - **Tomcat 9.0.24+**
+ - **Maven 3.5.4+**
+ 
+Hyperdrive-trigger can be packaged as a Web Application Archive and executed in web server.
```
Comment (kevinwallimann) [inline README.md]: Article: in _a_ web server

Comment (jozefbakus) [reply README.md]: Done, thanks

File: README.md (hunk 16 above)
Comment (jozefbakus) [reply README.md]: Done

File: README.md (hunk 15 above)
Comment (jozefbakus) [reply README.md]: yes you are right, done

File: README.md (hunk 1 above)
Comment (jozefbakus) [reply README.md]: Done

File: README.md (hunk 2 above)
Comment (jozefbakus) [reply README.md]: Done

File: README.md (hunk 3 above)
Comment (jozefbakus) [reply README.md]: Done

File: README.md (hunk 4 above)
Comment (jozefbakus) [reply README.md]: Done

Comment (jozefbakus) [reply README.md]: Done

File: README.md (hunk 14 above)
Comment (jozefbakus) [reply README.md]: Fixed, hmmm I am not sure, ` local and iterative` is worth of mention but fast is not that important

File: README.md (hunk 13 above)
Comment (jozefbakus) [reply README.md]: Done

File: README.md (hunk 12 above)
Comment (jozefbakus) [reply README.md]: Done

File: README.md (hunk 9 above)
Comment (jozefbakus) [reply README.md:58]: I think yes, I find it useful

File: README.md (hunk 8 above)
Comment (jozefbakus) [reply README.md]: Done

File: README.md (hunk 7 above)
Comment (jozefbakus) [reply README.md]: I am going to remove + .

File: README.md (hunk 6 above)
Comment (jozefbakus) [reply README.md]: Fixed.
Fixed.
No there is separate section for screenshots, I would have to create sections in User interface.

File: README.md (hunk 5 above)
Comment (jozefbakus) [reply README.md]: Done

File: README.md (hunk 11 above)
Comment (jozefbakus) [reply README.md]: It is just enumeration that is required for that way of building

File: README.md (hunk 10 above)
Comment (jozefbakus) [reply README.md]: Done

File: README.md [hunk 18]
```diff
@@ -52,7 +52,7 @@ Tested with:
 
 # How to build and run
 ## Application properties
-Before building application, update of application properties is required. Application properties templeate file could be find at `src/main/resources/application.properties`. Application properties that could be adjusted: 
+Adjusted application properties have to be provided. Application properties template file could be find at `src/main/resources/application.properties`. Application properties that could be adjusted:
```
Comment (kevinwallimann) [inline README.md]: _An_ application properties template file _can be found_ at ...
Application properties that _can_ be adjusted

File: README.md (hunk 9 above)
Comment (kevinwallimann) [reply README.md:58]: Ok, but we will have to keep the documentation in mind, whenever we make changes to the `application.properties` file

File: README.md (hunk 11 above)
Comment (kevinwallimann) [reply README.md]: My concern is that we have to change the same information in two places. Also, I believe you don't need npm preinstalled. Maven should install npm and node.

Comment (jozefbakus) [reply README.md]: I am going to remove it. Btw npm and angular is required because of `npm install` or `npm start`

File: README.md (hunk 9 above)
Comment (jozefbakus) [reply README.md:58]: Ok

File: README.md (hunk 18 above)
Comment (jozefbakus) [reply README.md]: Done

**Review by kevinwallimann — APPROVED** [review]
